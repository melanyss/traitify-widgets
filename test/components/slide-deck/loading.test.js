import Component from "components/slide-deck/loading";
import ComponentHandler from "support/component-handler";

describe("Loading", () => {
  let props;

  beforeEach(() => {
    props = {
      imageLoading: true,
      retry: jest.fn().mockName("retry"),
      translate: jest.fn().mockName("translate").mockImplementation((value) => value)
    };
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders retry button if not loading", () => {
    props.imageLoading = false;
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
