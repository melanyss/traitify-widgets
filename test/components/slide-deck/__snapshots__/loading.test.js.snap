// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders component 1`] = `
<div>
  <div>
    <div>
      <i />
      <i />
    </div>
  </div>
</div>
`;

exports[`Loading renders retry button if not loading 1`] = `
<div>
  <div>
    <div>
      <div>
        slide_error
      </div>
      <button
        onClick={[MockFunction retry]}
        type="button"
      >
        try_again
      </button>
    </div>
  </div>
</div>
`;
